pkg/controller/networkpolicy/crd_utils.go

```diff
@@ -131,6 +136,36 @@ func (n *NetworkPolicyController) createAddressGroupForCRD(peer secv1alpha1.Netw
 	return normalizedUID
 }
 
+// createAddressGroupForClusterGroupCRD creates an AddressGroup object corresponding to a
+// ClusterGroup spec. If the AddressGroup already exists, it returns the key
+// otherwise it copies the ClusterGroup CRD contents to an AddressGroup resource and returns
+// its key.
+func (n *NetworkPolicyController) createAddressGroupForClusterGroupCRD(cg *v1alpha2.ClusterGroup) string {
+	// Find the internal Group corresponding to this ClusterGroup
+	igKey := internalGroupKeyFunc(cg)
+	ig, found, _ := n.internalGroupStore.Get(igKey)
+	if !found {
+		klog.V(2).Infof("Internal group %s not found.", igKey)
+		return ""
+	}
+	intGrp := ig.(*antreatypes.Group)
+	// Check to see if the AddressGroup already exists
+	_, found, _ = n.addressGroupStore.Get(igKey)
+	if found {
+		return igKey
+	}
+	// Create an AddressGroup object for this internal Group.
+	addressGroup := &antreatypes.AddressGroup{
+		UID:          types.UID(igKey),
+		Name:         igKey,
+		Selector:     intGrp.Selector,
+		GroupMembers: intGrp.GroupMembers,
+	}
+	klog.V(2).Infof("Creating new AddressGroup %s with selector (%s) corresponding to ClusterGroup CRD", addressGroup.Name, addressGroup.Selector.NormalizedName)
+	n.addressGroupStore.Create(addressGroup)
```

Comment:
How does clusterGroup update trigger its corresponding AddressGroup update?

Reply:
by enqueuing the key(key is same for AddressGroup and corresponding internal Group) to addressGroup work queue. actually i added that line now after internal group is synced